plugins/flatpak/gs-flatpak.c

```diff
@@ -2567,6 +2567,72 @@ gs_flatpak_related_should_download (GsFlatpak *self, GsApp *app, FlatpakRelatedR
 		}
 	}
 
+	/* EknServices apps should not be downloaded in an "Endless Platform"
+	 * update if no other installed apps use that branch of the runtime */
+	if (g_str_has_prefix (name, "com.endlessm.EknServices")) {
+		g_autoptr(GPtrArray) xrefs_installed = NULL;
+		g_autoptr(GError) error = NULL;
+		const gchar *services_name_ver;
+		gboolean found_app_that_uses_runtime = FALSE;
+
+		services_name_ver = name + 24;  /* strlen("com.endlessm.EknServices") */
+		if (*services_name_ver == '\0')
+			services_name_ver = "1";
+
+		xrefs_installed = flatpak_installation_list_installed_refs_by_kind (self->installation,
+		                                                                    FLATPAK_REF_KIND_APP,
+		                                                                    NULL, &error);
+		if (xrefs_installed == NULL) {
+			g_debug ("Couldn't get installed apps (%s), adding %s anyway",
+			         error->message, name);
+			return TRUE;
+		}
+
+		for (guint i = 0; i < xrefs_installed->len; i++) {
+			FlatpakInstalledRef *xref = g_ptr_array_index (xrefs_installed, i);
+			g_autoptr(GBytes) data = NULL;
+			g_autoptr(GsApp) runtime = NULL;
+			const gchar *str;
+			gsize len;
+
+			data = flatpak_installed_ref_load_metadata (xref, NULL, &error);
+			if (data == NULL) {
+				g_warning ("Couldn't get metadata for %s (%s), ",
+				           flatpak_ref_get_name (FLATPAK_REF (xref)),
+				           error->message);
+				g_clear_error (&error);
+				continue;
+			}
+
+			str = g_bytes_get_data (data, &len);
+			runtime = gs_flatpak_create_runtime_from_metadata (self,
+									   str, len,
+									   &error);
+			if (runtime == NULL) {
+				g_warning ("Couldn't get runtime for %s (%s)",
+				           flatpak_ref_get_name (FLATPAK_REF (xref)),
+				           error->message);
+				g_clear_error (&error);
+				continue;
+			}
+
+			/* Special case for EknServices 1 and eosplatform 3.1 */
+			if (g_strcmp0 (services_name_ver, "1") == 0 &&
+			    g_strcmp0 (gs_app_get_id (app), "com.endlessm.Platform") == 0 &&
+			    g_strcmp0 (gs_app_get_branch (app), "eos3.1")) {
```

Comment:
Should be `g_strcmp0 (...) == 0`, no? (If not, prefer the explicit `!= 0` notation.)